### PR TITLE
Add ws-sec related testcases

### PR DESCRIPTION
**Pull request by chanikag**
This is to add testcases for https://github.com/wso2/api-manager/issues/1013

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: integration/mediation-tests/tests-platform/tests-userstore/src/test/java/org/wso2/micro/integrator/ws/policies/SecureProxyTestCase.java [hunk 1]
```diff
@@ -153,12 +151,12 @@ public void testSecureProxyEnableOnlyHTTPS() throws Exception {
         assertEquals(symbol, "WSO2", "Fault: value 'symbol' mismatched");
 
     }
-
+    //
```
Comment (GDLMadushanka) [inline integration/mediation-tests/tests-platform/tests-userstore/src/test/java/org/wso2/micro/integrator/ws/policies/SecureProxyTestCase.java:154]: empty comment

File: integration/mediation-tests/tests-platform/tests-userstore/src/test/java/org/wso2/micro/integrator/ws/policies/SecureProxyTestCase.java [hunk 2]
```diff
@@ -40,16 +38,16 @@ public class SecureProxyTestCase extends ESBIntegrationTest {
...
-        applySecurity();
+        super.init(TestUserMode.SUPER_TENANT_ADMIN);
         secureAxisServiceClient = new SecureServiceClient();
-
+        SampleAxis2Server axis2Server = new SampleAxis2Server();
+        axis2Server.deployService(SampleAxis2Server.SIMPLE_STOCK_QUOTE_SERVICE);
+        axis2Server.start();
     }
 
     @AfterClass(alwaysRun = true)
```
Comment (GDLMadushanka) [inline integration/mediation-tests/tests-platform/tests-userstore/src/test/java/org/wso2/micro/integrator/ws/policies/SecureProxyTestCase.java:53]: should we stop the axis2Server here ?

File: integration/mediation-tests/tests-platform/tests-userstore/src/test/resources/security/policies/scenario-config.xml [hunk 3]
```diff
@@ -0,0 +1,201 @@
+<!--
+  ~ Copyright 2022 WSO2, Inc. (http://wso2.com)
```
Comment (GDLMadushanka) [inline integration/mediation-tests/tests-platform/tests-userstore/src/test/resources/security/policies/scenario-config.xml:2]: LLC

File: integration/mediation-tests/tests-platform/tests-userstore/src/test/java/org/wso2/micro/integrator/ws/policies/SecureProxyTestCase.java [hunk 4]
```diff
@@ -170,50 +168,7 @@ public void testSecureProxyEnableOnlyHTTP() throws Exception {
...
-
-    private void clearUploadedResource()
-            throws InterruptedException, ResourceAdminServiceExceptionException, RemoteException {
-
-        ResourceAdminServiceClient resourceAdminServiceStub = new ResourceAdminServiceClient(
-                contextUrls.getBackEndUrl(), sessionCookie);
-
-        resourceAdminServiceStub.deleteResource("/_system/config/policy");
-        resourceAdminServiceStub.deleteResource("/_system/config/proxy");
+    private void clearUploadedResource() throws IOException, AutomationUtilException {
```
Comment (sajinieKavindya) [inline integration/mediation-tests/tests-platform/tests-userstore/src/test/java/org/wso2/micro/integrator/ws/policies/SecureProxyTestCase.java:171]: Seems this method does actually nothing. So, can we get rid of this?

File: integration/mediation-tests/tests-platform/tests-userstore/src/test/resources/security/policies/scenario1-policy.xml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+<!--
+  ~ Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (sajinieKavindya) [inline integration/mediation-tests/tests-platform/tests-userstore/src/test/resources/security/policies/scenario1-policy.xml:2]: Let's update the license headers with LLC. Please refactor other places as well.

File: integration/mediation-tests/tests-platform/tests-userstore/src/test/resources/security/policies/scenario6-policy.xml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            <sp:Trust10>
+                <wsp:Policy>
+                    <sp:MustSupportIssuedTokens/>
+                    <sp:RequireClientEntropy/>
+                    <sp:RequireServerEntropy/>
+                </wsp:Policy>
+            </sp:Trust10>
+        </wsp:All>
+    </wsp:ExactlyOne>
+</wsp:Policy>
```
Comment (sajinieKavindya) [inline integration/mediation-tests/tests-platform/tests-userstore/src/test/resources/security/policies/scenario6-policy.xml:75]: Let's add a newline at the end of the file here

File: integration/mediation-tests/tests-platform/tests-userstore/src/test/resources/security/policies/scenario8-policy.xml [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            <sp:Trust10>
+                <wsp:Policy>
+                    <sp:MustSupportIssuedTokens/>
+                    <sp:RequireClientEntropy/>
+                    <sp:RequireServerEntropy/>
+                </wsp:Policy>
+            </sp:Trust10>
+        </wsp:All>
+    </wsp:ExactlyOne>
+</wsp:Policy>
```
Comment (sajinieKavindya) [inline integration/mediation-tests/tests-platform/tests-userstore/src/test/resources/security/policies/scenario8-policy.xml:85]: Add a newline at here too

**Review by GDLMadushanka — APPROVED** [review]

**Pull request merged**